File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const erea = brown + yellow
    const factors = [];
    for(let i = 2; i <= Math.floor(Math.sqrt(erea)); i++){
        if(erea % i === 0) factors.push(i)
    }
    const filtered = factors.filter(el => (el - 2) * (erea / el - 2) === yellow);
    return [erea / filtered[0], filtered[0]]
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    area = brown + yellow
    factors = [];
    for i in range(2, int(area ** 0.5) + 1):
        if area % i == 0:
            if (i - 2) * (area / i - 2) == yellow:
                factors.append(area // i)
                factors.append(i)
    return factors